src/FxCop/Desktop.Analyzers/Core/Security/DoNotUseInsecureCryptographicAlgorithmsAnalyzer.cs

```diff
@@ -117,6 +118,28 @@ public void AnalyzeNode(SyntaxNodeAnalysisContext context)
             {
                 SyntaxNode node = context.Node;
                 SemanticModel model = context.SemanticModel;
+                string owningParentName = string.Empty;
+                SyntaxNode cur = node;
+
+                while(cur.Parent != null)
```

Comment:
I can only find GetEnclosingSymbol(), and it doesn't work for 

`await Task.Run(()=>new RC2CryptoServiceProvider());`